### PR TITLE
chore(svelte): Refactor infinity query implementation

### PR DESCRIPTION
**Pull request by fkling**
In preparation for other work, this commit substantially refactors the "infinity query" store implementation. The internals have been changed completely which allows us to simplify its public API.

- Simpler configuration. Query responses and next variables are now processed and merged in a single function.
- Restoration support. So far pages/components had to implement restoring the state of an infinity store on their own. Now the restoration strategy is part of the configuration. Pages/components only have to get an opaque snapshot via `store.capture()` and restore it via `store.restore(snapshot)`.
- More predictable state. It wasn't always obvious if the store content stale data e.g. during restoring data. Now `data` will only be set when the data is 'fresh'.
- Smarter 'incremental restoration' strategy. This strategy makes multiple requests to restore the previous state. It makes multiple requests because normally requests are cached and there this is fast. When the data is not cached though there is a noticable delay due to waterfall requests. Now we use a simple heuristic to determine whether or not GraqhQL data might be cached. If not we make a single request to restore the state.

For review I suggest to turn whitespace changes off.

## Test plan

Manual testing, unit tests.

### CODE REVIEW

**Comment by fkling** [comment]
Looks like type inference doesn't work as desired anymore with a single `mapResult` method. I'll probably split this up again into `map` and `combine`. Adding type declarations to these is a bit annoying IMO.

File: client/web-sveltekit/src/lib/repo/HistoryPanel.svelte [hunk 1]
```diff
@@ -1,76 +1,64 @@
...
+        await history.restore(data.history)
+
+        // If the selected revision is not in the set of currently loaded commits, load more
+        if (selectedRev) {
+            await history.fetchWhile(data => !data.find(commit => selectedRev?.startsWith(commit.abbreviatedOID)))
+        }
+
         if (data.scroller) {
-            // Wait until DOM was update before updating the scroll position
-            await tick()
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/repo/HistoryPanel.svelte:39]: Oh, haha. I just [asked](https://github.com/sourcegraph/sourcegraph/pull/63724#discussion_r1678504054) about this in my other PR.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte [hunk 2]
```diff
@@ -91,28 +90,15 @@
...
 
     $: ({ revision = '', parentPath, repoName, resolvedRevision, isCodyAvailable } = data)
     $: fileTreeStore.set({ repoName, revision: resolvedRevision.commitID, path: parentPath })
-    $: commitHistoryQuery = data.commitHistory
-    $: if (!!commitHistoryQuery) {
-        // Reset commit history when the query observable changes. Without
-        // this we are showing the commit history of the previously selected
-        // file/folder until the new commit history is loaded.
-        commitHistory = null
-    }
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte:106]: Nice!

**Review by camdencheek — APPROVED** [review]
This is great! This seems to address a big chunk of feedback I had after using `InfinityQuery` for the new references panel

**Pull request merged**